openlibrary/plugins/openlibrary/js/service-worker.js

```diff
@@ -30,75 +40,117 @@ registerRoute(
             }),
-            new CacheableResponsePlugin({
-                statuses: [0, 200],
-            })
+            cacheableResponses
+        ],
+    })
+)
+
+registerRoute(
+    /\/static\/build/,
+    new CacheFirst({
+        cacheName: 'static-cache-build',
+        plugins: [
+            new ExpirationPlugin({
+                maxAgeSeconds: 60 * 5,
+                purgeOnQuotaError: true,
+            }),
+            cacheableResponses
+        ],
+    })
+)
+
+/*
+Covers:
+S/M covers - cache 150 of them to take up no more than 2.25mb of space.
+L covers - cache 5 of them but with a very short timeout so if you're looking at a few they don't keep loading but they take up much of your storage.
+Original covers (with no letter) - save as above, just in case
+*/
+
+// Neither of these have the starting / because they don't come from the same origin
+// These urls could be domain agnostic if we want them to work locally but it would make them less readable
+const smallMediumCovers = new RegExp('.+(S|M).jpg');
+// This includes large covers and original size because of the ? in "L?"
+const largeCovers = new RegExp('.+L?.jpg');
+registerRoute(
+    smallMediumCovers,
+    new CacheFirst({
+        cacheName: 'covers-cache-small-medium',
+        plugins: [
+            new ExpirationPlugin({
+                maxEntries: 150,
+                purgeOnQuotaError: true,
+            }),
+            cacheableResponses
         ],
     })
 );
 
-// assets cache
 registerRoute(
-    /\.(?:js|css|woff)/,
-    new NetworkFirst({
-        cacheName: 'assets-cache',
+    largeCovers,
+    new CacheFirst({
+        cacheName: 'covers-cache-large',
+        plugins: [
+            new ExpirationPlugin({
+                maxEntries: 5,
+                maxAgeSeconds: daySeconds / 24,
+                purgeOnQuotaError: true,
+            }),
+            cacheableResponses
+        ],
     })
 );
 
-// images cache (precache)
+// cache all other images that happen to be on the page
+// This will only be same origin
 registerRoute(
     /\.(?:png|jpg|jpeg|svg|gif)/,
     new CacheFirst({
         cacheName: 'image-cache',
         plugins: [
             new ExpirationPlugin({
-                maxAgeSeconds: 7 * 24 * 60 * 60,
+                maxEntries: 50,
+                maxAgeSeconds: 7 * daySeconds,
+                purgeOnQuotaError: true
             })
         ],
     })
 );
 
-// others page html cache
+// cache all other requests on the same origin
 registerRoute(
-    new RegExp('/.+'),
+    /\/*/,
     new NetworkFirst({
         cacheName: 'other-html-cache',
         plugins: [
             new ExpirationPlugin({
-                // Due to redirect keep the value 2 times.
-                maxEntries: 10,
-                // Don't keep any entries for more than 30 days.
-                maxAgeSeconds: 30 * 24 * 60 * 60,
-                // Automatically cleanup if quota is exceeded.
+                maxEntries: 150,
```

Comment:
```suggestion
                maxEntries: 50,
```

But let's revisit on testing.